pyproject.toml

```diff
@@ -88,13 +88,13 @@ pyiceberg-core = { version = ">=0.5.1,<0.7.0", optional = true }
 polars = { version = "^1.21.0", optional = true }
 thrift-sasl = { version = ">=0.4.3", optional = true }
 kerberos = {version = "^1.3.1", optional = true}
-datafusion = { version = ">=45", optional = true }
+datafusion = { version = ">=45,<49", optional = true }
```

Comment:
i tested locally that this passes without the extra `<49`
```
poetry lock
make install
poetry run pytest tests/table/test_datafusion.py 
```

Reply:
its fine to leave as is. due to limitations we actually cannot use datafusion 49, so this makes sense